### PR TITLE
`Integrated code lifecycle`: Always add tag to docker image

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

If no tag is specified, the docker java API will pull all images with that name. This PR fixes that by always specifying a tag.
### Description

I added a new method that fixes that.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise 

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Programming Exercise with custom build plan and remove the tag
4. Observe your Artemis instance only pulling and executing one image

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [DockerConfig.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6060/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.aeolus/DockerConfig.html) | 100% | ✅ |
| [LocalCITriggerService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6060/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.localci/LocalCITriggerService.html) | 90% | ✅ |

## Summary by CodeRabbit

- **New Features**
	- Enhanced Docker configuration to ensure full image names, including tags, are properly formatted and utilized within the system.
	- Modified functionality to retrieve Docker image names from a `windfile` metadata object using a new method call `getFullImageName()` on the `Docker` object.
	- Updated assertion for `dockerImage` comparison in `LocalCIIntegrationTest.java` to check for `fullImageName` instead of just `image`.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/DockerConfig.java [hunk 1]
```diff
@@ -46,4 +46,19 @@ public List<String> getParameters() {
...
+     */
+    public String getFullImageName() {
+        if (tag == null) {
+            if (!image.contains(":")) {
+                return image + ":" + "latest";
+            }
+            return image;
+        }
+        return image + ":" + tag;
+    }
```
[the remark this answers is not in the document]
Comment (reschandreas) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/DockerConfig.java:63]: why would that improve the contains call?

[the remark this answers is not in the document]
Comment (reschandreas) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/DockerConfig.java:63]: a trim won't change if a ":" is in the string or not

**Review by Strohgelaender — APPROVED** [review]
looks reasonable. I also don't think that you should add a `trim()` call :D

**Review by rstief — APPROVED** [review]
code looks good 👍

**Review by laurenzfb — APPROVED** [review]
code lgtm. Thanks for fixing this :)

**Review by mateusmm01 — APPROVED** [review]
Code lgtm!

File: src/test/java/de/tum/in/www1/artemis/connectors/AeolusTest.java [hunk 2]
```diff
@@ -155,4 +155,17 @@ void testAeolusRepository() {
...
+        DockerConfig dockerConfig = new DockerConfig();
+        dockerConfig.setImage("image");
+        dockerConfig.setTag("tag");
+        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:tag");
+        dockerConfig.setTag(null);
+        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:latest");
+        dockerConfig.setImage("image:tag");
+        dockerConfig.setTag("notshown");
+        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:tag");
+    }
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/connectors/AeolusTest.java:170]: Please separate the cases like this:
```suggestion
    void testImageTagCombinations() {
        DockerConfig dockerConfig = new DockerConfig();
        dockerConfig.setImage("image");
        dockerConfig.setTag("tag");
        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:tag");
        
        dockerConfig.setTag(null);
        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:latest");
        
        dockerConfig.setImage("image:tag");
        dockerConfig.setTag("notshown");
        assertThat(dockerConfig.getFullImageName()).isEqualTo("image:tag");
    }
```

Comment (reschandreas) [reply src/test/java/de/tum/in/www1/artemis/connectors/AeolusTest.java:170]: will do!

**Review by Strohgelaender — APPROVED** [review]

**Review by rstief — APPROVED** [review]
re-approve after test changes

**Review by laurenzfb — APPROVED** [review]
reapprove after magic changes

**Pull request merged**